### PR TITLE
Expose the operation name from `juniper_rocket::GraphQLRequest`

### PR DESCRIPTION
**Pull request by davidpdrsn**
> Measuring the runtime of queries will only tell if there are slow
> queries. To find out which queries are slow you need the operation name.
> 
> Getting the operation name was previously not possible from a Rocket
> request handler. This fixes that.

I ran into this issue recently when implementing performance performance
monitoring for an app using juniper-rocket.

### CODE REVIEW

**Comment by theduke** [comment]
Looks good.

Two nits: 

* Formatting is failing (please run `cargo fmt` on **stable**)
* also update the `juniper` changelog

**Comment by davidpdrsn** [comment]
@theduke Done

File: juniper_rocket/src/lib.rs [hunk 1]
```diff
@@ -107,6 +107,19 @@ where
...
     }
+
+    pub fn operation_name(&self) -> Option<&str> {
+        match self {
+            GraphQLBatchRequest::Single(req) => req.operation_name(),
+            GraphQLBatchRequest::Batch(reqs) => {
+                if reqs.len() == 1 {
+                    reqs.get(0).and_then(|req| req.operation_name())
+                } else {
+                    None
```
Comment (LegNeato) [inline juniper_rocket/src/lib.rs]: Wouldn't we want a list of operations for batch requests for the same usecase?

Comment (davidpdrsn) [reply juniper_rocket/src/lib.rs]: Yeah that is true. Fixed in https://github.com/graphql-rust/juniper/pull/353/commits/168a763253476d4cc72ac20a8d2b03dab89dfd94

**Comment by davidpdrsn** [comment]
`cargo test` fails for me on nightly, which is necessary for rocket to work. Not quite sure what is wrong. I get this output

```
$ cargo test
warning: /Users/davidpdrsn/dev/major/juniper-fork/integration_tests/juniper_tests/Cargo.toml: file found to be present in multiple build targets: /Users/davidpdrsn/dev/major/juniper-fork/integration_tests/juniper_tests/src/lib.rs
   Compiling juniper_hyper v0.2.0 (/Users/davidpdrsn/dev/major/juniper-fork/juniper_hyper)
error: internal compiler error: src/librustc_mir/borrow_check/nll/universal_regions.rs:741: cannot convert `ReScope(Node(257))` to a region vid

thread 'rustc' panicked at 'Box<Any>', src/librustc_errors/lib.rs:637:9
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
error: aborting due to previous error


note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.36.0-nightly (3f5152e20 2019-05-08) running on x86_64-apple-darwin

note: compiler flags: -C debuginfo=2 -C incremental --crate-type bin

note: some of the compiler flags provided by cargo are hidden

error: Could not compile `juniper_hyper`.
warning: build failed, waiting for other jobs to finish...
error: internal compiler error: src/librustc_mir/borrow_check/nll/universal_regions.rs:741: cannot convert `ReScope(Node(257))` to a region vid

thread 'rustc' panicked at 'Box<Any>', src/librustc_errors/lib.rs:637:9
note: Run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
error: aborting due to previous error


note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.36.0-nightly (3f5152e20 2019-05-08) running on x86_64-apple-darwin

note: compiler flags: -C debuginfo=2 -C incremental

note: some of the compiler flags provided by cargo are hidden

error: Could not compile `juniper_hyper`.

To learn more, run the command again with --verbose.
```

Exact toolchain is `rustc 1.36.0-nightly (3f5152e20 2019-05-08)`

**Comment by LegNeato** [comment]
The bug you are hitting is https://github.com/rust-lang/rust/issues/59344.

**Comment by davidpdrsn** [comment]
It seems a [solution](https://github.com/rust-lang/rust/pull/60765) was recently merged. I'll give it another shot on the latest nightly.

**Comment by davidpdrsn** [comment]
I have [added an integration](https://github.com/graphql-rust/juniper/pull/353/commits/84c410dae5e8af5872ea33fcdb2b24a247563a29) test now. I wasn't quite sure if it also made sense to test the path where there is no operation name provided. I decided to trust the typesystem 😊 Let me know if you would like that.

**Comment by LegNeato** [comment]
Thanks!

**Pull request merged**